docs/developing.rst

```diff
@@ -0,0 +1,350 @@
+* You can only access the plugin and verify the correctness of your functionality via REST client externally. 
+
+Here is a sample for integration test for your reference:
+
+.. code:: java
+
+ public class XXXIT extends SQLIntegTestCase { // Extends our base test class
+ 
+     @Override
+     protected void init() throws Exception {
+         loadIndex(Index.ACCOUNT); // Load predefined test index mapping and data
+     }
+ 
+     @Override
+     public void testXXX() { // Test query against the index and make assertion
+         JSONObject response = executeQuery("SELECT ...");
+         Assert.assertEquals(6, getTotalHits(response));
+     }
+ }
+
+Finally thanks to JaCoCo library, you can check out the test coverage for your changes easily.
+
+Deploying Locally
+-----------------
+
+Sometime you want to deploy your changes to local Elasticsearch cluster, basically there are couple of steps you need to follow:
+
+1. Re-assemble to generate plugin jar file with your changes.
+2. Replace the jar file with the new one in your workspace.
+3. Restart Elasticsearch cluster to take it effect.
+
+
+To automate this common task, you can prepare an all-in-one command for reuse. Below is a sample command for MacOS::
+
+ ./gradlew assemble && {echo y | cp -f build/distributions/opendistro_sql-1*0.jar <Elasticsearch_home>/plugins/opendistro-sql} && {kill $(ps aux | awk '/[E]lasticsearch/ {print $2}'); sleep 3; nohup <Elasticsearch_home>/bin/elasticsearch > ~/Temp/es.log 2>&1 &}
+
+Note that for the first time you need to create `opendistro-sql` folder and unzip `build/distribution/opendistro_sql-xxxx.zip` to it.
+
+
+Documentation
+=============
+
+Dev Docs
+--------
+
+For new feature or big enhancement, it is worth document your design idea for others to understand your code better. There is already a docs/dev folder for all this kind of development documents.
+
+Reference Manual
+----------------
+
+Currently the reference manual documents are generated from a set of special integration tests. The integration tests use custom DSL to build ReStructure Text markup with real query and result set captured and documented.
+
+1. Add a new template to `src/test/resources/doctest/templates`.
+2. Add a new test class as below with `@DocTestConfig` annotation specifying template and test data used.
+3. Run `./gradlew build` to generate the actual documents into `docs/user` folder.
+
+Sample test class:
+
+.. code:: java
+
+ @DocTestConfig(template = "interfaces/protocol.rst", testData = {"accounts.json"})
+ public class ProtocolIT extends DocTest {
+ 
+     @Section(1)
+     public void test() {
+         section(
+             title("A New Section"),
+             description(
+                 "Describe what is the use of new functionality."
+             ),
+             example(
+                 description("Describe what is the use case of this example to show"),
+                 post("SELECT ...")
+             )
+         );
+     }
```

Comment:
Not sure if you meant to close out the class here, but we're missing a }.

Reply:
Good catch. I missed it. :)